src/providers/vod/vod-provider.ts

```diff
@@ -35,16 +37,27 @@ export class VodProvider extends Provider {
       this._eventManager.listen(this._player, this._player.Event.TEXT_TRACK_CHANGED, this._handleLanguageChange);
     }
     if (this._isPreventSeekActive) {
-      this._eventManager.listen(this._player, this._player.Event.TIME_UPDATE, () => this._onTimeUpdate());
+      this._eventManager.listen(this._player, this._player.Event.TIME_UPDATE, this._onTimeUpdate);
     }
   }
 
-  private _onTimeUpdate(): void {
-    for (const cp of this._cuePointsData) {
-      const cpToAdd = cp.filter((c: { startTime: number; }) => c.startTime <= this._player.currentTime && c.startTime >= this._lastTimeUpdated);
-      this._addCuePointToPlayer(cpToAdd);
+  private _onTimeUpdate = (): void => {
+    if (this._maxCurrentSeekTime < this._player.currentTime) {
+      this._maxCurrentSeekTime = this._player.currentTime; // Update max current seek time
+    }
+    if (this._player.currentTime - this._lastTimeCuePointsPushed >= TIME_UPDATE_DELTA) {
```

Comment:
so on forward seek this should be fine ?

Reply:
you can't seek forward :)